packages/finance/src/coin/mod.rs

```diff
@@ -40,10 +39,14 @@ where
         }
     }
 
-    pub fn is_zero(&self) -> bool {
+    pub const fn is_zero(&self) -> bool {
         self.amount == Zero::ZERO
     }
 
+    pub const fn into_amount(self) -> Amount {
+        self.amount
+    }
+
```

Comment:
please remove it, I understand the motivation ... but that should not happen at this price

Reply:
We don't have a way to neither do arithmetic, nor use trait methods in constant time. What do you suggest that I do?